tests/test_models.py

```diff
@@ -302,3 +302,81 @@ def test_should_cleanup_job(
                 s in caplog.text
                 for s in REANA_RUNTIME_KUBERNETES_KEEP_ALIVE_JOBS_WITH_STATUSES
             )
+
+
+@pytest.mark.parametrize(
+    "running_workflows, REANA_MAX_CONCURRENT_BATCH_WORKFLOWS,  priority",
+    [
+        (2, 10, 0.8),
+        (3, 10, 0.7),
+        (4, 10, 0.6),
+        (15, 30, 0.5),
+        (12, 30, 0.6),
+        (17, 50, 0.66),
+        (7, 64, 0.89),
+    ],
+)
+def test_get_workflow_overload_priority(
+    db,
+    session,
+    new_user,
+    running_workflows,
+    REANA_MAX_CONCURRENT_BATCH_WORKFLOWS,
+    priority,
+):
+    """Test logic to determine workflow overload priority factor based on running workflows."""
+    with mock.patch(
+        "reana_db.models.REANA_MAX_CONCURRENT_BATCH_WORKFLOWS",
+        REANA_MAX_CONCURRENT_BATCH_WORKFLOWS,
+    ):
+        for _ in range(running_workflows):
+            workflow_name = "test-workflow"
+            workflow = Workflow(
+                id_=str(uuid4()),
+                name=workflow_name,
+                owner_id=new_user.id_,
+                reana_specification=[],
+                type_="serial",
+                logs="",
+                status=RunStatus.running,
+            )
+            session.add(workflow)
+            session.commit()
+
+        assert new_user.get_workflow_overload_priority() == priority
+
+
+@pytest.mark.parametrize(
+    "complexity,priority,cluster_memory",
+    [
+        ([(1, 8589934592.0), (2, 4294967296.0), (5, 4294967296.0)], 55, 85899345920.0),
+        ([(2, 8)], 96, 400),
+        ([(1, 6), (1, 6)], 97, 400),
+        ([(5, 2), (5, 2)], 95, 400),
+        ([(1, 396)], 1, 400),
+        ([(1, 401)], 0, 400),
+        ([], 0, 100),
+        ([(3, 5), (1, 5)], 80, 100),
+        ([(3, 20), (2, 10)], 20, 100),
+        ([(2, 20), (5, 1)], 55, 100),
+        ([(1, 1), (1, 1)], 98, 100),
+        ([(1, 10), (1, 20), (1, 5), (1, 5), (1, 10), (1, 10)], 40, 100),
+    ],
+)
+def test_get_workflow_complexity_priority(
+    db, session, new_user, complexity, priority, cluster_memory
+):
+    """Test ``get_complexity_priority``."""
+    workflow_name = "test-workflow"
+    workflow = Workflow(
+        id_=str(uuid4()),
+        name=workflow_name,
+        owner_id=new_user.id_,
+        reana_specification=[],
+        type_="serial",
+        logs="",
+        status=RunStatus.running,
+    )
```

Comment:
Minor: I see that there is a `run_workflow()` fixture defined in `reana-db`, have you tried to use it?

Reply:
Good point, I hadn't seen that fixture. I've changed it together with `test_get_workflow_overload_priority` and it simplifies the tests a lot, thanks!